API For Objectives

jvanecek opened the pull request:
First implementation assuming key results are empty

jvanecek committed:
feat: api for objectives with no key results

serpi90 commented on source/OKR-Model/ObjectiveRESTfulController.class.st:
```diff
@@ -0,0 +1,129 @@
+}
+
+{ #category : #'instance creation' }
+ObjectiveRESTfulController class >> within: aPeriodManagementSystem [ 
```
anObjectiveManagementSystem

serpi90 commented on source/OKR-Tests/ObjectiveRESTfulControllerTest.class.st:
```diff
@@ -0,0 +1,231 @@
+]
+
+{ #category : #'private - support' }
+ObjectiveRESTfulControllerTest >> requestToUpdateObjective: aPeriodLocation with: anUpdatePeriod [
```
Rename aPeriodLocation and anUpdatePeriod to reference Objective instead of Period

serpi90 reviewed (changes requested):
- Rename arguments referencing period.
- Add /objectives entry point to launcher tests

nchillo committed:
Delete de KeyResultManagementSystem and ObjectiveManagementSystem. New OKRManagementSystem that manage objectives and key results.
Changes in the way it manage objectives.

serpi90 committed:
refactor: recategorize packages and code formatting

nchillo committed:
Update to stargate release candidate. Change objective API to include key results. Add key result comparison and json parser/dumper.

nchillo committed:
commit provisorio del key result.

serpi90 committed:
refactor: formatting and fix tests

- Formatted with ba-st formatting.
- Changed testControllers to test that all implemented endpoints are deployed.
- Changed ObjectiveResfulController >> #objectiveWithOneKeyResultJson using a string representing a json inside a json, which was incorrect.
- Fixed missing systems and controllers on command line handler

serpi90 committed:
fix: improve coverage

- Rename OKRRRESTfulAPIServerRunnerTest to OKRCommandLineHandlerTest
- test Specification routes
